### ISSUE TITLE
Creating seed checkpoint assertion

### ISSUE THREAD

**Issue by rualark** [pytorch/torchtitan#1107]
Here: https://github.com/pytorch/torchtitan/blob/2f78d7b7824aa466428ea54d5edfb44c813a1b61/torchtitan/train.py#L464C9-L467C89

Should it be == 1:
```
            assert int(
                os.environ["WORLD_SIZE"]
            ) == 1, "Must create seed checkpoint using a single device, to disable sharding."
```

**Comment by tianyu-l** [comment]
Would you be interested in submitting a fix?
also in https://github.com/pytorch/torchtitan/blob/main/torchtitan/experiments/flux/train.py#L218
